src/Bicep.Core/Navigation/IArtifactReferenceSyntax.cs

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using Bicep.Core.Syntax;
+
+namespace Bicep.Core.Navigation;
+
+/// <summary>
+/// Objects that implement IArtifactReferenceSyntax, contain syntax that can reference a foregin artifact, the artifact address
+/// is returned by `TryGetPath` and `SourceSyntax` contains the source syntax object to use for error propagation.
```

Comment:
When do we have a syntax node that may or may not refer to something external, and how do we tell that at the syntactic level?

Reply:
in the case of modules you know if its external by the prefix `br`, `ts` vs a local path, the same would apply for providers I assume

Reply:
I think we're using "external" to mean different things here. (I meant "external to the template in which the statement appears," which was also the meaning of "foreign" in `IForeignTemplateReference`.) 

I left this comment on the PR because I think the code comment is incorrect. `SourceSyntax` will always be populated for given syntax types, and `TryGetPath()` will always have a non-null return value for a well-formed statement. Whether it returns `null` or not depends on whether the statement was syntactically valid, not whether the path refers to an artifact on a separate system.